src/main/java/io/adobe/cloudmanager/CloudManagerApiException.java

```diff
@@ -85,6 +88,12 @@ public CloudManagerApiException(ErrorType type, String baseUrl, String apiPath,
    */
   public CloudManagerApiException(ErrorType type, String... vars) {
     this.message = String.format(type.message, (Object[]) vars);
+    this.cause = null;
+  }
+
+  @Override
+  public ApiException getCause() {
```

Comment:
This makes the API even more dependent on Swagger internal classes. That makes switching to another client generator much harder in the future. I would recommend to expose as less as possible Swagger specific classes.

Reply:
The _entire_ solution is based on the swagger generated classes. Every API response is either a Swagger class, or a wrapper around a swagger class.

Reply:
There is IMHO a difference between model classes (which are mainly based on the spec which is stable) and the ApiException (which is a Java-Client specific implementation detail of the Swagger generator)